controllers/space/space_controller.go

```diff
@@ -340,7 +340,10 @@ func (r *Reconciler) manageNSTemplateSet(logger logr.Logger, space *toolchainv1a
 			}
 			nsTmplSet = NewNSTemplateSet(memberCluster.OperatorNamespace, space, spaceBindings.Items, tmplTier)
 			if err := memberCluster.Client.Create(context.TODO(), nsTmplSet); err != nil {
-				logger.Error(err, "failed to create NSTemplateSet on target member cluster")
+				if errors.IsAlreadyExists(err) {
```

Comment:
dumb question: how can the `nstemplateset` exist immediately after the `client.Get` failed to find it? Is there any other controller that creates `nstemplateset` resources or concurrent process of this controller ?

Reply:
not a dumb question! the NSTemplateSet is directly created on the target member cluster whereas this code runs on the host cluster, so I suspect it's a race condition

Reply:
Ha I was rereading your description of the PR, so you think it get's created but the cache for the member client in the host operator it's not yet updated with this new object , if I well understood ?

Reply:
yes, that's my assumption, indeed 🤷‍♂️